### ISSUE TITLE
Customize loadOnDemand field name

### ISSUE THREAD

**Issue by treker** [cheton/infinite-tree#30]
API data structure may have different field name, need a way to customize it. So we don't need to change API field name to 'loadOnDemand'

**Comment by cheton** [comment]
How about adding a "shouldLoadNodes" option, which will default to:

```js
shouldLoadNodes: function(node) {
     return !node.hasChildren() && node.loadOnDemand;
}
```

You can write a custom function to return a boolean value to determine whether to load child nodes when opening a node.

**Comment by cheton** [comment]
Fixed in [v1.16.0](https://github.com/cheton/infinite-tree/releases/tag/v1.16.0)

```js
var tree = new InfiniteTree({
    el: document.querySelector('#tree'),
    data: [
        { // node
            id: '<unique-node-id>', // Required
            name: 'Node 0',
            loadNodes: true
        }
    ],
    shouldLoadNodes: function(parentNode) {
        return !parentNode.hasChildren() && parentNode.loadNodes;
    },
    loadNodes: function(parentNode, next) {
        var nodes = [
            {
                id: 'node1',
                name: 'Node 1'
            },
            {
                id: 'node2',
                name: 'Node 2'
            }
        ];
        setTimeout(function() {
            next(null, nodes, function() {
                // Completed
            });
        }, 1000);
    }
});
```
